src/Core/Containers/VectorArray.hpp

```diff
@@ -3,20 +3,35 @@
 #include <Core/Containers/AlignedStdVector.hpp>
 #include <Core/RaCore.hpp>
 
+#include <type_traits> // std::is_integral
+
 namespace Ra {
 namespace Core {
 
+/// Internal class used to compute types and values used by VectorArray, to handle both
+/// eigen matrices and integral types
+template <typename V, bool isIntegral, bool isEigen>
+struct VectorArrayTypeHelper {};
+
 /// This class is a wrapper around a std::vector of Core::Vectors.
 /// which allow to use the stdlib's dynamic array implementation, yet pass it as
 /// a matrix when Eigen needs it with the getMap() method.
 template <typename V>
 class VectorArray : public AlignedStdVector<V>
 {
+  private:
+    using TypeHelper =
+        VectorArrayTypeHelper<V,
+                              std::is_arithmetic<V>::value,
+                              std::is_base_of<typename Eigen::MatrixBase<V>, V>::value>;
+
   public:
     // Type shortcuts
-    using Vector         = V;
-    using Matrix         = Eigen::Matrix<typename V::Scalar, V::RowsAtCompileTime, Eigen::Dynamic>;
-    using MatrixMap      = Eigen::Map<Matrix>;
+    static constexpr int NumberOfComponents = TypeHelper::NumberOfComponents;
+    // scalar_type is different from value_type defined in Base class
+    using scalar_type = typename TypeHelper::scalar_type;
+    using Matrix      = Eigen::Matrix<scalar_type, TypeHelper::NumberOfComponents, Eigen::Dynamic>;
```

Comment:
Is there a mean to disable this declaration (and others) for non eigen type ? using std::enable°if or something like this ? I'm not sure (and do not success in my writing tests but ...)

Reply:
Matrix typedef is valid for non eigen types, as long as you canmap data to eigen matrix. Check via getMatrix availability